lib/src/client.dart

```diff
@@ -516,6 +516,14 @@ class SubiquityClient {
       AdPasswordValidation.values.byName,
     );
   }
+
+  Future<AdJoinResult> joinActiveDirectory({bool wait = true}) async {
```

Comment:
Don't you feel like that method name doesn't convey an idea of taking an action?

Reply:
how would you prefer to name it? :) `getActiveDirectoryJoinResult()`?

Reply:
Yeap! Sounds better. We are just monitoring a result of an action initiated by somebody else.